sunpy/map/compositemap.py

```diff
@@ -431,14 +433,22 @@ def plot(self, axes=None, annotate=True,  # pylint: disable=W0613
             }
             params.update(matplot_args)
 
+            # The request to show a map layer rendered as a contour is indicated by a
+            # non False levels property.  If levels is False, then the layer is
+            # rendered using imshow.
             if m.levels is False:
-                ret.append(axes.imshow(m.data, **params))
+                # Check for the presence of masked map data
+                if m.mask is None:
+                    ret.append(axes.imshow(m.data, **params))
+                else:
+                    ret.append(axes.imshow(np.ma.array(np.asarray(m.data), mask=m.mask), **params))
+            else:
+                # Check for the presence of masked map data
+                if m.mask is None:
+                    ret.append(axes.contour(m.data, m.levels, **params))
+                else:
+                    ret.append(axes.contour(np.ma.array(np.asarray(m.data), mask=m.mask), **params))
```

Comment:
If `levels` exist, but there's also a mask... then the levels are not being used for the contour?

Reply:
Ah, good catch!  Thanks!